app/controllers/api/v1/orders_controller.rb

```diff
@@ -56,20 +69,33 @@ def recent_designations
       render json: orders
     end
 
+    def my_orders
+      render json: @orders, each_serializer: serializer, root: "orders",
+        include_packages: false, browse_order: true
+    end
+
     private
 
     def order_record
-      @order = Order.accessible_by(current_ability).where(stockit_id: order_params[:stockit_id]).first_or_initialize
-      @order.assign_attributes(order_params)
-      @order.stockit_activity = stockit_activity
-      @order.stockit_contact = stockit_contact
-      @order.stockit_organisation = stockit_organisation
-      @order.detail = stockit_local_order
+      if order_params[:stockit_id]
+        @order = Order.accessible_by(current_ability).where(stockit_id: order_params[:stockit_id]).first_or_initialize
+        @order.assign_attributes(order_params)
+        @order.stockit_activity = stockit_activity
+        @order.stockit_contact = stockit_contact
+        @order.stockit_organisation = stockit_organisation
+        @order.detail = stockit_local_order
+      elsif is_browse_app
+        @order.assign_attributes(order_params)
+        @order.created_by = current_user
+        @order.detail_type = "GoodCity"
+      end
       @order
     end
 
     def order_params
-      params.require(:order).permit(:stockit_id, :code, :status, :created_at, :stockit_contact_id, :detail_id, :detail_type, :stockit_organisation_id, :description, :stockit_activity_id)
+      params.require(:order).permit(:stockit_id, :code, :status, :created_at,
+        :stockit_contact_id, :detail_id, :detail_type, :description, :state, :state_event, :stockit_organisation_id, :stockit_activity_id, :purpose_description,
```

Comment:
Technically, it would be possible for a user to change their own order status here. We might want to restrict that further down the line.